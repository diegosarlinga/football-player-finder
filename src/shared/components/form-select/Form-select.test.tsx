import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import FormSelect from './Form-select';

describe('FormSelect component tests', () => {
  let wrapper: ShallowWrapper;

  describe('when rendered with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<FormSelect/>);
    });
  
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when rendered with props', () => {
    let onChangeSpy: jest.Mock;
    beforeEach(() => {
      onChangeSpy = jest.fn();
      const props = {
        label: 'some label',
        value: 'some value',
        name: 'some name',
        options: ['opt 1', 'opt 2'],
        className: 'test-class',
        onChange: onChangeSpy,
      };
      wrapper = shallow(<FormSelect {...props}/>);
    });
  
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should emit when select change', () => {
      wrapper.find('select').simulate('change');

      expect(onChangeSpy).toHaveBeenCalledTimes(1);
    });

  });
});
