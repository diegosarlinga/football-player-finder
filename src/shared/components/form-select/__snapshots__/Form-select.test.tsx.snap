// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect component tests when rendered with no props should match snapshot 1`] = `
<div
  className="Form-select-container form-group w-100 undefined"
>
  <label
    className="w-100"
  >
    <select
      className="form-control"
    >
      <option
        key="0"
      />
    </select>
  </label>
</div>
`;

exports[`FormSelect component tests when rendered with props should match snapshot 1`] = `
<div
  className="Form-select-container form-group w-100 test-class"
>
  <label
    className="w-100"
  >
    some label
    <select
      className="form-control"
      name="some name"
      onChange={[MockFunction]}
      placeholder="some label"
      value="some value"
    >
      <option
        key="0"
      />
      <option
        key="0"
        value="opt 1"
      >
        opt 1
      </option>
      <option
        key="1"
        value="opt 2"
      >
        opt 2
      </option>
    </select>
  </label>
</div>
`;
